VSTSRestApiSamples/GettingStarted/Authentication.cs

```diff
@@ -1,55 +1,203 @@
+            }
+
+            var bearerAuthHeader = new AuthenticationHeaderValue("Bearer", result.AccessToken);
+
+            return ListProjects(vstsAccountName, bearerAuthHeader);
+        }
+
+        public ListofProjectsResponse.Projects NonInteractivePersonalAccessToken(string vstsAccountName, string personalAccessToken)
+        {
             // encode our personal access token                   
-            string credentials = Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes(string.Format("{0}:{1}", "", personalAccessToken)));
+            string encodedPAT = Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes(string.Format("{0}:{1}", "", personalAccessToken)));
+            var basicAuthHeader = new AuthenticationHeaderValue("Basic", encodedPAT);
+
+            return ListProjects(vstsAccountName, basicAuthHeader);
+        }
+
+        public ListofProjectsResponse.Projects DeviceCodeADAL(string vstsAccountName, string applicationId)
+        {
+            string tenant = GetAccountTenant(vstsAccountName);
+            AuthenticationContext ctx = GetAuthenticationContext(tenant);
+
+            AuthenticationResult result = null;
+            try
+            {
+                DeviceCodeResult codeResult = ctx.AcquireDeviceCodeAsync(VSTSResourceId, applicationId).Result;
+                Console.WriteLine("You need to sign in.");
+                Console.WriteLine("Message: " + codeResult.Message + "\n");
+                result = ctx.AcquireTokenByDeviceCodeAsync(codeResult).Result;
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+
+            var bearerAuthHeader = new AuthenticationHeaderValue("Bearer", result.AccessToken);
+
+            return ListProjects(vstsAccountName, bearerAuthHeader);
+        }
 
+        private static ListofProjectsResponse.Projects ListProjects(string vstsAccountName, AuthenticationHeaderValue authHeader)
+        {
             // create a viewmodel that is a class that represents the returned json response
             ListofProjectsResponse.Projects viewModel = new ListofProjectsResponse.Projects();
 
             // use the httpclient
             using (var client = new HttpClient())
             {
-                client.BaseAddress = new Uri(url);  // url of our account (https:// accountname.visualstudio.com)
+                client.BaseAddress = new Uri(String.Format("https://{0}.visualstudio.com", vstsAccountName));
                 client.DefaultRequestHeaders.Accept.Clear();
                 client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
-                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", credentials); 
+                client.DefaultRequestHeaders.Add("User-Agent", "VstsRestApiSamples");
+                client.DefaultRequestHeaders.Add("X-TFS-FedAuthRedirect", "Suppress");
```

Comment:
I would add a doc reference but it doesn't seem to be documented anywhere on VS.com or MSDN